pkg/liqoctl/install/aks/provider.go

```diff
@@ -0,0 +1,215 @@
+	if err != nil {
+		return err
+	}
+
+	if err = k.parseClusterOutput(ctx, &cluster); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateChartValues patches the values map with the values required for the selected cluster.
+func (k *aksProvider) UpdateChartValues(values map[string]interface{}) {
+	values["apiServer"] = map[string]interface{}{
+		"address": k.endpoint,
+	}
+	values["networkManager"] = map[string]interface{}{
+		"config": map[string]interface{}{
+			"serviceCIDR": k.serviceCIDR,
+			"podCIDR":     k.podCIDR,
+		},
+	}
+}
+
+// GenerateFlags generates the set of specific subpath and flags are accepted for a specific provider.
+func GenerateFlags(flags *flag.FlagSet) {
+	subFlag := flag.NewFlagSet(providerPrefix, flag.ExitOnError)
+	subFlag.SetNormalizeFunc(func(f *flag.FlagSet, name string) flag.NormalizedName {
+		return flag.NormalizedName(installutils.PrefixedName(providerPrefix, name))
+	})
+
+	subFlag.String("subscription-id", "", "The ID of the Azure Subscription of your cluster (optional)")
+	subFlag.String("subscription-name", "", "The Name of the Azure Subscription of your cluster (optional)")
+	subFlag.String("resource-group-name", "", "The Azure ResourceGroup name of your cluster")
+	subFlag.String("resource-name", "", "The Azure Name of your cluster")
+
+	flags.AddFlagSet(subFlag)
+}
+
+func (k *aksProvider) parseClusterOutput(ctx context.Context, cluster *containerservice.ManagedCluster) error {
+	switch cluster.NetworkProfile.NetworkPlugin {
+	case containerservice.NetworkPluginKubenet:
+		k.setupKubenet(cluster)
+	case containerservice.NetworkPluginAzure:
+		if err := k.setupAzureCNI(ctx, cluster); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unknown AKS network plugin %v", cluster.NetworkProfile.NetworkPlugin)
+	}
+
+	k.endpoint = *cluster.Fqdn
+
+	return nil
+}
+
+// setupKubenet setups the data for a Kubenet cluster.
+func (k *aksProvider) setupKubenet(cluster *containerservice.ManagedCluster) {
+	k.podCIDR = *cluster.ManagedClusterProperties.NetworkProfile.PodCidr
+	k.serviceCIDR = *cluster.ManagedClusterProperties.NetworkProfile.ServiceCidr
+}
+
+// setupAzureCNI setups the data for an Azure CNI cluster.
+func (k *aksProvider) setupAzureCNI(ctx context.Context, cluster *containerservice.ManagedCluster) error {
+	vnetSubjectID := (*cluster.AgentPoolProfiles)[0].VnetSubnetID
+
+	networkClient := network.NewSubnetsClient(k.subscriptionID)
+	networkClient.Authorizer = *k.authorizer
+
+	vnetName, subnetName, err := parseSubnetID(*vnetSubjectID)
+	if err != nil {
+		return err
+	}
+
+	vnet, err := networkClient.Get(ctx, k.resourceGroupName, vnetName, subnetName, "")
+	if err != nil {
+		return err
+	}
+
+	k.podCIDR = *vnet.AddressPrefix
+	k.serviceCIDR = *cluster.ManagedClusterProperties.NetworkProfile.ServiceCidr
+
+	return nil
+}
+
+func parseSubnetID(subnetID string) (vnetName, subnetName string, err error) {
+	strs := strings.Split(subnetID, "/")
```

Comment:
I think this may require some commentary and may be a util method to be less obscure.